### DISCUSSION TITLE
Transforming bvecs after registration

### DISCUSSION THREAD

**Discussion by weningerleon** [dipy/dipy#2089] [Registration] [answered]
Hi,

I'd like to co-register two different raw diffusion images with an affine transformation.
Currently, I'm registering these using FA maps, and then applying the transform to all diffusion images.
How do I then apply the transform to the bvecs? I couldn't find any function for this in dipy.
My best guess would be this, but as I can't really verify the result, I would be very grateful for a second opinion:

    bvecs_new = np.matmul(affine[0:3,0:3], bvecs.T).T

**Comment by arokem** [comment, marked as answer]
Does this do what you want? https://github.com/dipy/dipy/blob/master/dipy/core/gradients.py#L447-L501

**Comment by weningerleon** [comment]
Perfect, I didn't find this function.
Thanks!
